### ISSUE TITLE
Doc site redesign & design guidelines

### ISSUE THREAD

**Issue by leeejune** [AlaskaAirlines/AuroDocsSite#93]
## Is your feature request related to a problem? Please describe.
Our doc site lacks design guidelines (do's/don'ts, use cases, accessibility information, anatomy, etc) for designers to refer to.

## Describe the solution you'd like
Add the following things for each component:
- Anatomy
- Variants
- Behaviors
- Best practices
	- Do's and Don'ts
	- Use cases
- Accessibility
	- Color
	- Interaction
	- Keyboard interaction
- Content guidelines
- Checklist

## Additional context

After this work, I would also like to revamp the look of the doc site.
